### PR TITLE
Support new syntax for type alias with free parameters

### PR DESCRIPTION
**Pull request by yuyichao**

### CODE REVIEW

File: src/Compat.jl [hunk 1]
```diff
@@ -1815,7 +1817,7 @@ end
 if VERSION < v"0.5.0-dev+3669"
     using Base: promote_op
     import Base: *
-    typealias SubMatrix{T} SubArray{T,2}
+    eval(Expr(:typealias, :(SubMatrix{T}), :(SubArray{T,2})))
```
Comment (TotalVerb) [inline src/Compat.jl]: Why doesn't `@compat SubMatrix{T} = SubArray{T,2}` work here?

Comment (yuyichao) [inline src/Compat.jl]: Just avoiding using `@compat` in this file in general.

Comment (TotalVerb) [inline src/Compat.jl]: `@compat` is used 20 or so times in this file. Is there a reason to avoid it?

Comment (yuyichao) [inline src/Compat.jl]: I just don't ever want to worry about ordering unless it's very hard to do without `@compat`.

**Comment by yuyichao** [comment]
Added support for `@compat const A{T} = B{T}` since it might be useful to keep the style consistent...

**Comment by tkelman** [comment]
before anyone tags a new version here, should be sure to resolve https://github.com/JuliaLang/Compat.jl/pull/324#issuecomment-279244196 in the readme

**Pull request merged**